app/src/main/java/com/portfolio/tasky/entry/repositories/EntryRepository.kt

```diff
@@ -0,0 +1,70 @@
+package com.portfolio.tasky.entry.repositories
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.portfolio.tasky.entry.network.EntryApiCall
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+
+class EntryRepository @Inject constructor(private val entryApiCall: EntryApiCall)  {
+    private val registrationLiveData = MutableLiveData<Boolean?>()
+    private val loginLiveData = MutableLiveData<AuthenticationResponse?>()
+
+    fun doRegistration(taskyCallStatusCallback: TaskyCallStatus, registerModel: RegisterRequest) : LiveData<Boolean?>{
+
+        taskyCallStatusCallback.onRequestCallStarted()
```

Comment:
Since you'll be using coroutines, you won't need a callback like this one

You can just wait for the response, since there's no need for callbacks in coroutines

Reply:
Current implementation requires this

Reply:
Done

Reply:
I have kept the call back but switched to coroutines.